kubernetes/kubernetes.md

```diff
@@ -0,0 +1,74 @@
+# windows_exporter on Kubernetes
+
+With Kubernetes supporting HostProcess containers on Windows nodes (as of [v1.22](https://kubernetes.io/blog/2021/08/16/windows-hostprocess-containers/), it is useful to run this windows_exporter as a container on Windows to export metrics for your Prometheus implementation.
+
+Requirements:
+
+- Kuberentes 1.22+
+- containerd 1.6 Beta+
+- WindowsHostProcessContainers feature-gate (currently in alpha in 1.22) turned on for kube-apiserver and kubelet if using Kubernetes 1.22
+
+> IMPORTANT: This does not work unless you are specifically targeting Host Process Containers with Containerd (Docker doesn't have support).  The image will build and run but will be **not** able to access the host.
```

Comment:
Should we add an init container to the ds-yaml below that runs a check to make sure everything is configured properly to run host-process containers?
Something as simple as https://github.com/kubernetes/kubernetes/blob/6d9008b1b086008caed5a9d8ef150cca0abaefa9/test/e2e/windows/host_process.go#L109 should be sufficient.
I think this could prevent a lot of confusion / time spent debugging.

Reply:
I guess you'd have to this too and then make sure %COMPUTERNAME% in the container matches the node name env var 
https://github.com/kubernetes/kubernetes/blob/6d9008b1b086008caed5a9d8ef150cca0abaefa9/test/e2e/windows/host_process.go#L505-L513

Reply:
+1 for the daemon set yaml but wonder if, once you're on k8s then you know if its hostprocess or not anyway?  good one to think about.  Could this be done in the binary to check it has access to the requested counters at startup?

Reply:
or output running as a user in the log message?

Reply:
I like the idea of checking the current user and giving a warning in logs. 

I like the idea of adding an init container to the docs for hostprocess as a debugging tool but having across the board for projects seems a bit burdensome on the end user.

cc: @brasmith-ms on possible troubleshooting tips for hostprocess docs

Reply:
I found that with nanoserver, the container will not start due to https://github.com/prometheus-community/windows_exporter/blob/f84f54afdaa215259b443f2d6011c99902a181dd/exporter.go#L250 not having required libraries in nano.  This means if someone didn't add the `hostprocess` fields to the spec it would fail to start the container if using the image produced with the dockerfile here.  

Maybe we want to add a warning here too for usability?

With these updates, If someone built there own image and used servercore the container will start and get the following:

```
k logs windows-exporter-hbhvm                        
time="2021-12-01T21:07:39Z" level=warning msg="No filters specified for process collector. This will generate a very large number of metrics!" source="process.go:58"
time="2021-12-01T21:07:39Z" level=info msg="Running as User Manager\\ContainerAdministrator" source="exporter.go:354"
time="2021-12-01T21:07:39Z" level=warning msg="Running as a preconfigured Windows Container user. This may mean you do not have Windows HostProcess containers configured correctly and some functionality will not work as expected." source="exporter.go:356"
time="2021-12-01T21:07:39Z" level=info msg="Enabled collectors: process" source="exporter.go:359"
time="2021-12-01T21:07:39Z" level=info msg="Starting windows_exporter (version=latest, branch=use-hostprocess, revision=95e15738886f4ddd40cc277ae1821a0b0c6f5c0a)" source="exporter.go:411"
time="2021-12-01T21:07:39Z" level=info msg="Build context (go=go1.17.3, user=root@buildkitsandbox, date=20211201-20:43:54)" source="exporter.go:412"
time="2021-12-01T21:07:39Z" level=info msg="Starting server on :9182" source="exporter.go:415"
time="2021-12-01T21:07:39Z" level=info msg="TLS is disabled." source="gokit_adapter.go:38"

```